### ISSUE TITLE
[BUG][11.0] base: error with custom field

### ISSUE THREAD

**Issue by fmdl** [odoo/odoo#31447]
Impacted versions: 11.0

Steps to reproduce:
-create a custom selection field related and stored
-upgrade odoo with https://github.com/odoo/odoo/commit/5f5bb5e26b0cf49dc48d00eeac80fa25bca315c5
-starting raise

the issue come from that the field selection on ir.model.fields is set to `False`.

cc @Elkasitu

```
2019-02-27 00:21:10,971 60940 ERROR iguana_master werkzeug: Error on request:
Traceback (most recent call last):
  File "/Users/florentdelabarre/.virtualenvs/iguana/lib/python3.6/site-packages/werkzeug/serving.py", line 205, in run_wsgi
    execute(self.server.app)
  File "/Users/florentdelabarre/.virtualenvs/iguana/lib/python3.6/site-packages/werkzeug/serving.py", line 193, in execute
    application_iter = app(environ, start_response)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/service/server.py", line 260, in app
    return self.app(e, s)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/service/wsgi_server.py", line 166, in application
    return application_unproxied(environ, start_response)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/service/wsgi_server.py", line 154, in application_unproxied
    result = handler(environ, start_response)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/http.py", line 1319, in __call__
    return self.dispatch(environ, start_response)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/http.py", line 1293, in __call__
    return self.app(environ, start_wrapped)
  File "/Users/florentdelabarre/.virtualenvs/iguana/lib/python3.6/site-packages/werkzeug/wsgi.py", line 599, in __call__
    return self.app(environ, start_response)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/http.py", line 1474, in dispatch
    odoo.registry(db).check_signaling()
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/__init__.py", line 76, in registry
    return modules.registry.Registry(database_name)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/modules/registry.py", line 61, in __new__
    return cls.new(db_name)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/modules/loading.py", line 383, in load_modules
    registry.setup_models(cr)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/modules/registry.py", line 274, in setup_models
    model._setup_base()
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/models.py", line 2401, in _setup_base
    self.env['ir.model.fields']._add_manual_fields(self)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/addons/base/ir/ir_model.py", line 901, in _add_manual_fields
    field = self._instanciate(field_data)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/addons/base/ir/ir_model.py", line 892, in _instanciate
    attrs = self._instanciate_attrs(field_data)
  File "/Users/florentdelabarre/git/iguana-yachts/server/addons/mail/models/ir_model.py", line 70, in _instanciate_attrs
    attrs = super(IrModelField, self)._instanciate_attrs(field_data)
  File "/Users/florentdelabarre/git/iguana-yachts/server/addons/base_sparse_field/models/models.py", line 46, in _instanciate_attrs
    attrs = super(IrModelFields, self)._instanciate_attrs(field_data)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/addons/base/ir/ir_model.py", line 857, in _instanciate_attrs
    attrs['selection'] = safe_eval(field_data['selection'])
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/tools/safe_eval.py", line 348, in safe_eval
    c = test_expr(expr, _SAFE_OPCODES, mode=mode)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/tools/safe_eval.py", line 212, in test_expr
    pycompat.reraise(ValueError, ValueError('"%s" while compiling\n%r' % (ustr(e), expr)), exc_info[2])
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/tools/pycompat.py", line 86, in reraise
    raise value.with_traceback(tb)
  File "/Users/florentdelabarre/git/iguana-yachts/server/odoo/tools/safe_eval.py", line 206, in test_expr
    expr = expr.strip()
ValueError: "'NoneType' object has no attribute 'strip'" while compiling
None
```

**Comment by pedrobaeza** [comment]
The error doesn't seem to involve the changes in the mentioned commit.

**Comment by fmdl** [comment]
@pedrobaeza it is linked, because the mentioned commit set to `None` `selection`, then during `attrs['selection'] = safe_eval(field_data['selection'])` it fail.

**Comment by pedrobaeza** [comment]
OK, I see, then it should be put to `[]` instead AFAIK.

**Comment by Elkasitu** [comment]
I think the biggest problem here is that it uses a `safe_eval`, a `literal_eval` would probably work in the case of `None` and is safer, but that's something for master, I will make a fix for stable today

**Comment by Elkasitu** [comment]
@fmdl can you share how you created this field? a screenshot of it in DB? I cannot seem to reproduce the issue...

**Comment by fmdl** [comment]
@Elkasitu please find how :

- go to Odoo
- install stock
- create a new field from UI
- name = x_test
- type = selection
- model = stock.move
- related = picking_id.state
- selection = [('draft', 'Draft'),('waiting', 'Waiting Another Operation'),('confirmed', 'Waiting'), ('assigned', 'Ready'),('done', 'Done'),('cancel', 'Cancelled'),]
- save
-> Issue 1 : the field selection is now empty after te save

- upgrade stock
-> Issue 2 : raise

I think that the issue come from here: if there are nothing in the python it set None, but for custom field there are no python, it should take the value in the database.

can you reverse this comit, please. @nje-odoo @rco-odoo 

![image](https://user-images.githubusercontent.com/16716992/53492562-70e56580-3a99-11e9-9743-015c72ba5ad8.png)

**Comment by fmdl** [comment]
Note : on a custom related selection field, the field `selection options` show be readonly and take the value of the related field.

**Comment by Elkasitu** [comment]
Thanks, should be handled by #31466

**Comment by odony** [comment]
Fixed by reverting the original patch via #31466
